charm/src/iframe/recipe.ts

```diff
@@ -20,6 +21,7 @@ export const buildFullRecipe = (iframe: IFrameRecipe) => {
     argumentSchema: JSONSchema,
     resultSchema: JSONSchema,
     spec: string,
+    enhancedSpec?: string,
```

Comment:
let's store this

since we're not currently storing the goal, maybe we reuse the current `spec` for "enhancedSped" and add a `goal`, which is what this function gets as spec. So we should rename it all.